### ISSUE TITLE
[bug]  RCP getbalance("MyLabel") rong available balance

### ISSUE THREAD

**Issue by qlpqlp** [dogecoin/dogecoin#2701]
# Bug Report

**Current behavior:**
RCP getbalance("MyLabel") dosent show the correct balance, it shows the total recived.

**Expected behavior**
Has to show the available spendable balance.

**Steps to reproduce:**
curl --user myusername --data-binary '{"jsonrpc": "1.0", "id": "curltest", "method": "getbalance", "params": ["*"]}' -H 'content-type: text/plain;' http://127.0.0.1:22555/

Result: correct spendable balance
My use case: getbalance("*")) = 19.191142

curl --user myusername --data-binary '{"jsonrpc": "1.0", "id": "curltest", "method": "getbalance", "params": ["MyLabel"]}' -H 'content-type: text/plain;' http://127.0.0.1:22555/

Result: total recived amount insted of the spendable balance
My use case: getbalance("Dogecoin42069")) = 31.738042

My Adress with the label "Dogecoin42069" that im testing: DTqAFgNNUgiPEfGmc4HZUkqJ4sz5vADd1n

**Screenshots**
![dogecoin_available](https://user-images.githubusercontent.com/5968076/142238295-3d923f99-f54d-4219-ab51-576d2fed1271.png)

**Dogecoin Core version**
V1.14.5.0-31afd1331

**Machine specs**
- OS: Windows 10
- CPU: Intel J4125 @ 2Ghz
- RAM: 8GB DDR
- Disk size: 250GB
- Disk Type (HD/SDD): SSD

**Extra information**
Much great day for u all! Such codding!  :) Thanks!

**Comment by jadijadi** [comment]
Hi there. 
May I work on this as my first contribution to DogeCoin? Have checked the source by the way.
It seems `rpcwallet.cpp` is responsible for this and the issue is two different method used:

1. when using *, the rpcwallet.cpp/getbalance calcualtes it correctly
2. when using a labe, system calls the wallet.cpp/GetAccountBalance which returns the wrong number

I would be happy to work on it if agreed.

**Comment by patricklodder** [comment]
Please do note that the accounts functionality is marked as depreciated because it has been broken since forever and needs an overhaul, which we will mostly inherit with 1.21. 

Please open a potential pull request early, even if it doesn't fully work, so that we can help to make decisions early and not waste your time.

**Comment by jadijadi** [comment]
An Update: worked on this for a couple of days. Learned about the source and wallets; but as @patricklodder commented, it's not easy to fix this.

To be honest, I even think that there is a logical problem here: When depositing, we are depositing TO an address and its fine to be able to check the TOTAL RECEIVED to an address. But when spending coins, we do not say "spend from this address" and because of that, asking "how much coins do I have still have in this specific address" is not useful. 

In other words, when receiving, addresses can be seen as separate entities but when spending, we are spending from our whole wallet. Therefor, my understanding is that this *depreciation* flag is appropriate for this RPC unless we document it as "if a label is provided, this RPC will return the total received doges by this address".

**Comment by SpaceAlphaNet** [comment]
`getBalance(label)` will show you the **local value**. local value can be anything really, you can manage it the way you 
 want and can make a mess with the `move` command or  `sendfrom <Account label> <to address> <amount>` (it is possible to use any label and get negative balance.  locally)

**Comment by SpaceAlphaNet** [comment]
Also if you decide to use accounting you must know:

The accounts (labels) balance received from `getBalance(label)` saved locally so the [Change Addresses](https://support.exodus.com/article/109-what-are-change-addresses)  will be completely messed up if you try to send transaction with the same wallet or private key.

Use the wallet on different machine and the Change address will be sent to a random label!

**Comment by etherx-dev** [comment]
Also can we somehow recompute the label because loading the wallet file in testnet and back on main tesults in blank address

**Comment by envisiondata** [comment]
Is this still open? If so I would like to work on it.

Please let me know.

**Comment by patricklodder** [comment]
@envisiondata - [the comment above](https://github.com/dogecoin/dogecoin/issues/2701#issuecomment-1069044837) by @SpaceAlphaNet disputes that this is a bug and outlines the correct usage of the functionality. I honestly think trying to change the behavior on the 1.14 branch will be a waste of your time, especially because with 1.21 we'll have sqlite-based "descriptor wallets" that provide a much better foundation for the future.

Note that about 5% of all issues I deal with are around bdb, but these easily takes up 20% of my time. So I'd recommend focusing on this when we release 1.21 and then making sure that the wallet functionality is clean and as easy to use as we can get it.
